src/libstd/sync/mutex.rs

```diff
@@ -339,14 +339,14 @@ impl StaticMutex {
     #[inline]
     pub fn lock(&'static self) -> LockResult<MutexGuard<()>> {
         unsafe { self.lock.lock() }
-        MutexGuard::new(self, &DUMMY.0)
+        unsafe { MutexGuard::new(self, &DUMMY.0) }
```

Comment:
This is another case that would likely benefit from merging the two unsafe blocks